tests/unit/test_spaces.cpp

```diff
@@ -163,6 +164,150 @@ static dist_func_t<double> IP_dist_funcs_2ExtResiduals[] = {
     FP64_InnerProductSIMD2ExtResiduals_AVX512};
 } // namespace spaces_test
 
+TEST_F(SpacesTest, fp32_L2_GetDistFunc) {
+    using namespace spaces_test;
+    using namespace spaces;
+    // fp32_L2_dist_funcs_array[i][j] is a pointer to a function that match CalculationGuideline[i]
+    // and architecture optimization Arch_Optimization[j]
+    static dist_func_t<float> *fp32_L2_dist_funcs_array[] = {
+        nullptr, L2_dist_funcs_16Ext, L2_dist_funcs_4Ext, L2_dist_funcs_16ExtResiduals,
```

Comment:
why the first entry is null?

Reply:
i find it less confusing then
`fp32_L2_dist_funcs_array[dim_opt + 1][arch]`